Oraimunka24015004/src/AlgorithmicRecap.java

```diff
@@ -0,0 +1,172 @@
+        }
+        return result;
+    }
+
+    boolean taskTwoIsPalindrome(String text) {
+        return text.equals(taskOneReverts(text));
+    }
+
+    boolean taskThreeIsAnagram(String text, String text2) {
+        Map<Character, Integer> map = getCharacters(text);
+        Map<Character, Integer> map2 = getCharacters(text2);
+        return map.equals(map2);
+    }
+
+    private HashMap<Character, Integer> getCharacters(String text) {
+        HashMap<Character, Integer> map = new HashMap<>();
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsKey(character)) {
+                map.put(character, map.get(character) + 1);
+            } else {
+                map.put(character, 1);
+            }
+        }
+        return map;
+    }
+
+    BigInteger taskFourCountNumbers(String text) {
+        BigInteger result = BigInteger.valueOf(0);
+        for (int i = 0; i < text.length(); i++) {
+            Character character = text.charAt(i);
+            if (Character.isDigit(character)) {
+                result = result.add(BigInteger.valueOf(Character.getNumericValue(character)));
+            }
+        }
+        return result;
+    }
+
+    int taskFiveGetFactorial(int number) {
+        int result = 1;
+        for (int i = 1; i <= number; i++) {
+            result *= i;
+        }
+        return result;
+    }
+
+    char taskSixGetUniqueChar(String text) {
+        Map<Integer, Character> map = new HashMap<>();
+        int key = 0;
+        char result = ' ';
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsValue(character)) {
+                map.put(key + 1, character);
+            } else {
+                map.put(key + 1, character);
+                result = character;
+            }
+        }
+        return result;
+    }
+
+    String taskSevenGetAllUniqueChar(String text) {
+        String result = "";
+        List<Character> blacklist = new ArrayList<>();
+        int counter = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            for (int j = i + 1; j < text.length(); j++) {
+                char tempChar = text.charAt(j);
+                if (tempChar == character) {
+                    counter++;
+                }
+            }
+            if (counter == 0 && !blacklist.contains(character)) {
+                result += character;
+            } else {
+                blacklist.add(character);
+                counter = 0;
+            }
+        }
+        return result;
+    }
+
+    boolean taskEightSameAmountOfAb(String text, String text2) {
+        return (countSimilarChar('a', text) == countSimilarChar('a', text2))
+                && countSimilarChar('b', text) == countSimilarChar('b', text2);
+    }
+
+    private int countSimilarChar(char getChar, String text) {
```

Comment:
`getChar` is a kinda shady naming and i would have no idea what it stands for if I didn't check